src/test/kotlin/org/jitsi/rtp/rtp/RedRtpPacketTest.kt

```diff
@@ -155,6 +156,187 @@ class RedRtpPacketTest : ShouldSpec() {
                 packet.payloadLength shouldBe 62
             }
         }
+        context("Parsing a RED packet with short blocks") {
+            context("Without reading the redundancy blocks") {
+                val buffer = ByteArray(
+                    redPacketBytesShortBlocks.size +
+                        RtpPacket.BYTES_TO_LEAVE_AT_START_OF_PACKET +
+                        Packet.BYTES_TO_LEAVE_AT_END_OF_PACKET
+                )
+                redPacketBytesShortBlocks.copyInto(buffer, RtpPacket.BYTES_TO_LEAVE_AT_START_OF_PACKET)
+                val packet = RtpRedPacket(
+                    buffer,
+                    RtpPacket.BYTES_TO_LEAVE_AT_START_OF_PACKET,
+                    redPacketBytesShortBlocks.size
+                )
+                packet.payloadType shouldBe 112
+                packet.sequenceNumber shouldBe 35006
+                packet.timestamp shouldBe 862774129L
+                packet.ssrc shouldBe 624250097L
+                packet.hasExtensions shouldBe true
+                packet.getHeaderExtension(1).shouldHaveId1AndLen1()
+                packet.payloadLength shouldBe 4 + 1 + 3 + 3 // block headers + block lengths
```

Comment:
Maybe put the initialization of `packet` and the basic checks on it one level up, so you don't have to repeat it?

Reply:
Isn't `decapsulate` a destructive change?